### PR TITLE
Run e2e-simple tests also for distroless images

### PR DESCRIPTION
**Pull request by kramerul**
This is a PR to activate the tests introduced in https://github.com/istio/istio/pull/14944.

We are not exactly sure, if this is the only place where we need to add this test.

### CODE REVIEW

**Comment by howardjohn** [comment]
I think we probably want this in `istio.istio.master.yaml` as an optional (non-required) (can be made required if its not flaky for ~a week or so) presubmit test. Right now this is in the release and marked as required, so if a change breaks it it won't fail until we try to release (a daily build most likely, so not that bad) and then block the release.

Also just need to make sure https://github.com/istio/istio/pull/14944 gets merged in first.

**Comment by kramerul** [comment]
We moved the job to `istio.istio.master.yaml` and made it optional.

@howardjohn We are not really sure, if it's OK to call `scripts/build_test_presubmit.sh` within this file.

File: prow/cluster/jobs/istio/istio/istio.istio.master.yaml
Comment (howardjohn) [inline prow/cluster/jobs/istio/istio/istio.istio.master.yaml]: Just based on it not being in the other ones, I am a bit suspicious of this. @icygalz is the expert on this so I'll let her take a look, otherwise lgtm

Comment (utka) [reply prow/cluster/jobs/istio/istio/istio.istio.master.yaml]: Why do you want to run this script?
https://github.com/istio-releases/pipeline/blob/master/scripts/build_test_presubmit.sh

Comment (kramerul) [reply prow/cluster/jobs/istio/istio/istio.istio.master.yaml]: @utka, the distroless docker images are build in `prow/istio-presubmit.sh` called from `build_test_presubmit.sh`. These images are required for this test.

**Review by howardjohn — APPROVED** [review]
/lgtm
Thanks!

**Comment by howardjohn** [comment]
@utka @icygalz can you take a look?

**Review by utka — APPROVED** [review]

**Pull request merged**
